### PR TITLE
Fix ZeroMQ communication backend

### PR DESCRIPTION
**Pull request by chris-durand**
This fixes several problems in the ZeroMQ backend. The header-only implementation is also moved to cpp files. Thus ZeroMQ and zmqpp will be required for building xpcc on hosted. Is this a problem?

### CODE REVIEW

**Review by salkinium — APPROVED** [review]

**Comment by salkinium** [comment]
> Thus ZeroMQ and zmqpp will be required for building xpcc on hosted. Is this a problem?

That may be ok, but better ask @strongly-typed.

**Comment by dergraaf** [comment]
@chris-durand Can you update the files with correct copyright information (not "Roboterclub Aachen" but real persons)?

**Comment by chris-durand** [comment]
> That may be ok, but better ask @strongly-typed 

He suggested me to add a unit test which would require ZeroMQ as a build dependency anyway.

**Comment by chris-durand** [comment]
> @chris-durand Can you update the files with correct copyright information (not "Roboterclub Aachen" but real persons)?

I added comments with author information to the header files.

File: src/xpcc/communication/xpcc/backend/zeromq/reader.hpp [hunk 1]
```diff
@@ -0,0 +1,177 @@
+// coding: utf-8
+/* Copyright (c) 2016, Roboterclub Aachen e. V.
```
Comment (ekiwi) [inline src/xpcc/communication/xpcc/backend/zeromq/reader.hpp]: @chris-durand this needs to be the actual person who created this code, German copyright law afaik does not allow you to assign copyright to a company

Comment (chris-durand) [reply src/xpcc/communication/xpcc/backend/zeromq/reader.hpp]: @ekiwi This line is in every source file of xpcc, I have just copied that.

Comment (ekiwi) [reply src/xpcc/communication/xpcc/backend/zeromq/reader.hpp]: yes, we recently discovered, that this is non ideal

this is also what @dergraaf was referring to in his comment about updating the copyright information.

Comment (salkinium) [reply src/xpcc/communication/xpcc/backend/zeromq/reader.hpp]: This should really be:
```diff
-/* Copyright (c) 2016, Roboterclub Aachen e. V.
+/* Copyright (c) 2016, Sascha Schade
+ * Copyright (c) 2017, Christopher Durand
```
We are going to [forward port these changes to modm](https://github.com/modm-io/modm/blob/develop/src/modm/communication/xpcc/backend/zeromq/connector.hpp#L2), which use the correct copyright information. So to make it a little easier for us, we are requesting this change.

**Comment by salkinium** [comment]
We are currently working on the next generation of this framework (over at [modm-io](https://github.com/modm-io), it's very much WIP), which will be licensed under MPLv2 (see below for why). Since this is a non-trivial change to xpcc, would you be ok with licensing your contributions as MPLv2 as well?

> We are relicensing all source code from 3-clause BSD in xpcc to MPLv2 in modm.
> https://www.mozilla.org/en-US/MPL/2.0/  
> https://en.wikipedia.org/wiki/Mozilla_Public_License
> 
> The MPLv2 license is more restrictive than the BSD license, but less restrictive than (L)GPL:
> - it requires copyleft at file level, but derivatives can be released under other licenses (that are compatible with MPL).
> - in contrast to GPL you only have to provide open-source access to the derived code covered under MPL.
> - It contains a patent retaliation clause, similar to that of the Apache license.
>
> We think MPLv2 is a better choice for modm, as it makes it easier for us to keep modm open-source (copyleft), but does _not_ deter programmers, projects or companies to use modm and contribute back like GPL does.

**Comment by chris-durand** [comment]
@salkinium I have added the requested copyright information and am ok with relicensing my contributions as MPLv2.

**Pull request merged**